### PR TITLE
Fix: TrueNAS Core support for pool stats

### PR DESCRIPTION
**Pull request by rgon10**
## Proposed change

Correct enablePools for API differences between TrueNAS Core and TrueNAS Scale. 

TrueNAS Scale returns allocated and free at the pool level for pools, where TrueNAS Core returns this information for the individual volumes below the pool. 

### Current /proxy?type=truenas response on Core
```json
{
  "id" : "pool-name",
  "name": "pool-name",
  "healthy": true,
}
```

### Current behavior when `enablePools: true` is set with TrueNAS Core
image

### Abbreviated output of pool API on TrueNAS Core

```json
"topology": {
  "data": [
      {
          "type": "RAIDZ2",
          "stats": {
              "size": 123456789123456,
              "allocated": 1234567879,
          },
      },
  ]},                     
```

### Overview
This fix takes the data response and adds it to the the /proxy/type=truenas response.
A second .jsx class is defined to support differentiation between core and scale. 
A default of scale is included to default to the existing behavior if `nasType` is not defined. 

## Type of change

- [ ] New service widget
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [ ] Other (please explain)

## Checklist:

- [x] If applicable, I have added corresponding documentation changes.
- [x] If applicable, I have reviewed the [feature](https://gethomepage.dev/latest/more/development/#new-feature-guidelines) and / or [service widget guidelines](https://gethomepage.dev/latest/more/development/#service-widget-guidelines).
- [x] I have checked that all code style checks pass using [pre-commit hooks](https://gethomepage.dev/latest/more/development/#code-formatting-with-pre-commit-hooks) and [linting checks](https://gethomepage.dev/latest/more/development/#code-linting).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

File: src/widgets/truenas/component.jsx [hunk 1]
```diff
@@ -39,9 +42,19 @@ export default function Component({ service }) {
         <Block label="truenas.alerts" value={t("common.number", { value: alertData.pending })} />
       </Container>
       {enablePools &&
+        scaleDeployment &&
```
Comment (shamoon) [inline src/widgets/truenas/component.jsx:45]:
```suggestion
        widget?.nasType === "scale" &&
```

This whole thing could be an if really, i.e. if enablePools --> scale vs else

File: src/widgets/truenas/component.jsx [hunk 2]
```diff
@@ -39,9 +42,19 @@ export default function Component({ service }) {
...
+          <ScalePool
+            key={pool.id}
+            name={pool.name}
+            healthy={pool.healthy}
+            allocated={pool.allocated}
+            free={pool.free}
+          />
         ))}
+      {enablePools &&
+        coreDeployment &&
```
Comment (shamoon) [inline src/widgets/truenas/component.jsx:56]:
```suggestion
        widget?.nasType === "core" &&
```

**Review by shamoon — CHANGES_REQUESTED** [review]
Thanks, a couple comments.

One other big thing is its very un-DRY to create the core-pool component. core-pool and scale-pool contain almost identical code so they should remain as one `pool.jsx` and you can pass in an extra parameter or something like that to do the small amount of extra logic required for core.

I also think this wont pass style checks as-is

Edit: does pass lint =)

**Comment by shamoon** [comment]
Also, if you wouldn't mind posting an example API output from your core device so I can at least test that out (and useful for future reference). Eg. https://github.com/gethomepage/homepage/pull/2908#issuecomment-1947759182

**Comment by rgon10** [comment]
Thank you for the initial review and the comments. Your comment about an if/else sparked an idea on how to handle the update in a cleaner manner. 

I've pushed an additional commit, which hopefully makes more sense than the two file approach. 

Please let me know if you have any additional suggestions. 

Full API output is below.

```
[
    {
        "id": 1,
        "name": "r510-hot",
        "guid": "6556082935218011584",
        "encrypt": 0,
        "encryptkey": "",
        "path": "/mnt/r510-hot",
        "status": "ONLINE",
        "scan": {
            "function": "SCRUB",
            "state": "FINISHED",
            "start_time": {
                "$date": 1711864806000
            },
            "end_time": {
                "$date": 1711933573000
            },
            "percentage": 99.87462759017944,
            "bytes_to_process": 47897138294784,
            "bytes_processed": 47957264142336,
            "bytes_issued": 47897138294784,
            "pause": null,
            "errors": 0,
            "total_secs_left": null
        },
        "topology": {
            "data": [
                {
                    "type": "RAIDZ2",
                    "path": null,
                    "guid": "1535553132487403937",
                    "status": "ONLINE",
                    "stats": {
                        "timestamp": 1215877933914035,
                        "read_errors": 0,
                        "write_errors": 0,
                        "checksum_errors": 0,
                        "ops": [
                            0,
                            1614206072,
                            242171406,
                            0,
                            0,
                            0,
                            0
                        ],
                        "bytes": [
                            0,
                            75253679603712,
                            5722574843904,
                            0,
                            0,
                            0,
                            0
                        ],
                        "size": 47828755808256,
                        "allocated": 32456332099584,
                        "fragmentation": 22,
                        "self_healed": 0,
                        "configured_ashift": 12,
                        "logical_ashift": 9,
                        "physical_ashift": 12
                    },
                    "children": [
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/6d24d093-8f61-11ec-904a-782bcb6f2f8e",
                            "guid": "8696719541792796358",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877934043103,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    133772879,
                                    26664693,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    9521700507648,
                                    717204451328,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da0p2",
                            "disk": "da0",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/2c4fb61c-7a61-11ed-87a9-782bcb6f2f8f",
                            "guid": "2947882732905558508",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877934124679,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    234243268,
                                    30802385,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    9347680493568,
                                    715055337472,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da10p2",
                            "disk": "da10",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/de5740eb-d600-11ee-9e4e-782bcb6f2f8e",
                            "guid": "6605496022463082297",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877934206813,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    192365816,
                                    29278130,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    9415930576896,
                                    715754217472,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da18p2",
                            "disk": "da18",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/0d9e9234-18ea-11ed-8718-782bcb6f2f8e",
                            "guid": "3505478959618909518",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877934293418,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    233344912,
                                    33170451,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    9362482143232,
                                    714111725568,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da6p2",
                            "disk": "da6",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/92ca9f85-7364-11ed-87a9-782bcb6f2f8f",
                            "guid": "16123964721279297651",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877934375831,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    233359691,
                                    30811583,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    9349646163968,
                                    715069579264,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da9p2",
                            "disk": "da9",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/0939bd03-5bab-11ed-87a9-782bcb6f2f8f",
                            "guid": "9873160099347274881",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877934458804,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    231545259,
                                    33116897,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    9365547057152,
                                    713764573184,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da8p2",
                            "disk": "da8",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/43c9ee58-913d-11ed-87a9-782bcb6f2f8f",
                            "guid": "14213741150302750027",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877934536189,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    245831443,
                                    31707520,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    9334585339904,
                                    714692829184,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da11p2",
                            "disk": "da11",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/6a904965-8f7f-11ec-904a-782bcb6f2f8e",
                            "guid": "4767329962964346060",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877934614971,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    109742804,
                                    26619747,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    9556107321344,
                                    716922130432,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da1p2",
                            "disk": "da1",
                            "unavail_disk": null
                        }
                    ],
                    "unavail_disk": null
                },
                {
                    "type": "RAIDZ2",
                    "path": null,
                    "guid": "8336907661775220706",
                    "status": "ONLINE",
                    "stats": {
                        "timestamp": 1215877935090177,
                        "read_errors": 0,
                        "write_errors": 0,
                        "checksum_errors": 0,
                        "ops": [
                            0,
                            342842753,
                            182957545,
                            0,
                            0,
                            0,
                            0
                        ],
                        "bytes": [
                            0,
                            36315078070272,
                            5061278330880,
                            0,
                            0,
                            0,
                            0
                        ],
                        "size": 27986006900736,
                        "allocated": 15500645928960,
                        "fragmentation": 19,
                        "self_healed": 0,
                        "configured_ashift": 12,
                        "logical_ashift": 9,
                        "physical_ashift": 12
                    },
                    "children": [
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/f815b8e7-99c5-11ed-ba8e-782bcb6f2f8e",
                            "guid": "4535096430269276220",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877935211982,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    41167801,
                                    25224905,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    5206785634304,
                                    723792936960,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da12p2",
                            "disk": "da12",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/d965926e-d690-11ee-9e4e-782bcb6f2f8e",
                            "guid": "7987842585236311595",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877935297748,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    91050292,
                                    24597584,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    5099829493760,
                                    723988328448,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da7p2",
                            "disk": "da7",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/ac9ce49a-99c5-11ed-ba8e-782bcb6f2f8e",
                            "guid": "8652142879031749570",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877935374574,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    39240650,
                                    25421885,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    5210998697984,
                                    723636129792,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da13p2",
                            "disk": "da13",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/fa22b98a-99c5-11ed-ba8e-782bcb6f2f8e",
                            "guid": "14073583766925637105",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877935453915,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    43913017,
                                    26988430,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    5197165899776,
                                    722433613824,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da14p2",
                            "disk": "da14",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/f9f99986-99c5-11ed-ba8e-782bcb6f2f8e",
                            "guid": "11063049712386002313",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877935534094,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    43419189,
                                    26905013,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    5198191300608,
                                    722483998720,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da15p2",
                            "disk": "da15",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/fa69d525-99c5-11ed-ba8e-782bcb6f2f8e",
                            "guid": "8432991220455819778",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877935615390,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    41863000,
                                    26898239,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    5201438883840,
                                    722467770368,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da16p2",
                            "disk": "da16",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/fa45e92d-99c5-11ed-ba8e-782bcb6f2f8e",
                            "guid": "1619293864639998190",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215877935696128,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    42188804,
                                    26921489,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    5200668160000,
                                    722475552768,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da17p2",
                            "disk": "da17",
                            "unavail_disk": null
                        }
                    ],
                    "unavail_disk": null
                }
            ],
            "log": [],
            "cache": [],
            "spare": [],
            "special": [],
            "dedup": []
        },
        "healthy": true,
        "status_detail": null,
        "autotrim": {
            "value": "off",
            "rawvalue": "off",
            "parsed": "off",
            "source": "DEFAULT"
        },
        "encryptkey_path": null,
        "is_decrypted": true
    },
    {
        "id": 2,
        "name": "r510-hv",
        "guid": "5800654724888980408",
        "encrypt": 0,
        "encryptkey": "",
        "path": "/mnt/r510-hv",
        "status": "ONLINE",
        "scan": {
            "function": "SCRUB",
            "state": "FINISHED",
            "start_time": {
                "$date": 1711872002000
            },
            "end_time": {
                "$date": 1711875850000
            },
            "percentage": 99.97553825378418,
            "bytes_to_process": 740630507520,
            "bytes_processed": 740811718656,
            "bytes_issued": 740630507520,
            "pause": null,
            "errors": 0,
            "total_secs_left": null
        },
        "topology": {
            "data": [
                {
                    "type": "RAIDZ2",
                    "path": null,
                    "guid": "4697684372707939453",
                    "status": "ONLINE",
                    "stats": {
                        "timestamp": 1215857727152289,
                        "read_errors": 0,
                        "write_errors": 0,
                        "checksum_errors": 0,
                        "ops": [
                            0,
                            34585608,
                            236357471,
                            0,
                            0,
                            0,
                            0
                        ],
                        "bytes": [
                            0,
                            2522230046720,
                            7890366341120,
                            0,
                            0,
                            0,
                            0
                        ],
                        "size": 4776003633152,
                        "allocated": 741195386880,
                        "fragmentation": 27,
                        "self_healed": 0,
                        "configured_ashift": 12,
                        "logical_ashift": 9,
                        "physical_ashift": 12
                    },
                    "children": [
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/46bb64ae-0c56-11ed-a4ce-782bcb6f2f8f",
                            "guid": "10857852813979877557",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215857727284990,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    8590652,
                                    59119745,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    630294028288,
                                    1975632773120,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da2p2",
                            "disk": "da2",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/46f53096-0c56-11ed-a4ce-782bcb6f2f8f",
                            "guid": "8373166459967436875",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215857727371314,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    8593441,
                                    59153422,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    629977481216,
                                    1972262608896,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da4p2",
                            "disk": "da4",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/46d9d21f-0c56-11ed-a4ce-782bcb6f2f8f",
                            "guid": "3718460956290198456",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215857727463785,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    8683850,
                                    59079213,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    630924767232,
                                    1974106791936,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da3p2",
                            "disk": "da3",
                            "unavail_disk": null
                        },
                        {
                            "type": "DISK",
                            "path": "/dev/gptid/471c6d53-0c56-11ed-a4ce-782bcb6f2f8f",
                            "guid": "9063503291394162883",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 1215857727552904,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    8717665,
                                    59005091,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    631033769984,
                                    1968364167168,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 12
                            },
                            "children": [],
                            "device": "da5p2",
                            "disk": "da5",
                            "unavail_disk": null
                        }
                    ],
                    "unavail_disk": null
                }
            ],
            "log": [],
            "cache": [],
            "spare": [],
            "special": [],
            "dedup": []
        },
        "healthy": true,
        "status_detail": null,
        "autotrim": {
            "value": "off",
            "rawvalue": "off",
            "parsed": "off",
            "source": "DEFAULT"
        },
        "encryptkey_path": null,
        "is_decrypted": true
    }
]
```

File: src/widgets/truenas/component.jsx [hunk 3]
```diff
@@ -40,7 +40,15 @@ export default function Component({ service }) {
       </Container>
       {enablePools &&
         poolsData.map((pool) => (
-          <Pool key={pool.id} name={pool.name} healthy={pool.healthy} allocated={pool.allocated} free={pool.free} />
+          <Pool
+            key={pool.id}
+            name={pool.name}
+            healthy={pool.healthy}
+            scaleAllocated={pool.allocated}
+            scaleFree={pool.free}
```
Comment (shamoon) [inline src/widgets/truenas/component.jsx:48]:
```suggestion
            allocated={pool.allocated}
            free={pool.free}
```

Just a nit to rename these

File: src/widgets/truenas/pool.jsx [hunk 4]
```diff
@@ -1,8 +1,19 @@
...
+  if (nasType === "scale") {
+    total = scaleFree + scaleAllocated;
+    allocated = scaleAllocated;
+  } else {
+    for (let i = 0; i < data.length; i += 1) {
+      total += data[i].stats.size;
+      allocated += data[i].stats.allocated;
+    }
+  }
+
```
Comment (shamoon) [inline src/widgets/truenas/pool.jsx:16]:
```suggestion
export default function Pool({ name, free, allocated, healthy, data, nasType }) {
  let total = 0;
  if (nasType === "scale") {
    total = free + allocated;
  } else {
    allocated = 0;  // eslint-disable-line no-param-reassign
    for (let i = 0; i < data.length; i += 1) {
      total += data[i].stats.size;
      allocated += data[i].stats.allocated;  // eslint-disable-line no-param-reassign
    }
  }

```

File: src/utils/config/service-helpers.js [hunk 5]
```diff
@@ -520,6 +521,7 @@ export function cleanServiceGroups(groups) {
         }
         if (type === "truenas") {
           if (enablePools !== undefined) cleanedService.widget.enablePools = JSON.parse(enablePools);
+          cleanedService.widget.nasType = nasType ?? "scale";
```
Comment (shamoon) [inline src/utils/config/service-helpers.js:524]:
```suggestion
          if (nasType !== undefined) cleanedService.widget.nasType = nasType;
```

Sorry for the second change here but I'd like to keep the default value encapsulated to the widget itself, not in service-helpers.

File: docs/widgets/services/truenas.md [hunk 6]
```diff
@@ -11,6 +11,8 @@ To create an API Key, follow [the official TrueNAS documentation](https://www.tr
 
 A detailed pool listing is disabled by default, but can be enabled with the `enablePools` option.
 
+TrueNAS Scale and Core APIs differ, the optional `nasType` parameter is needed if using the `enablePools` option.
+
```
Comment (shamoon) [inline docs/widgets/services/truenas.md:15]:
```suggestion
To use the `enablePools` option with TrueNAS Core, the `nasType` parameter is required.

```

File: docs/widgets/services/truenas.md [hunk 7]
```diff
@@ -19,4 +21,5 @@ widget:
   password: pass # not required if using api key
   key: yourtruenasapikey # not required if using username / password
   enablePools: true # optional, defaults to false
+  nasType: scale # defaults to scale, values are core or scale
```
Comment (shamoon) [inline docs/widgets/services/truenas.md:24]:
```suggestion
  nasType: scale # defaults to scale, must be set to 'core' if using enablePools with TrueNAS Core
```

File: src/widgets/truenas/component.jsx [hunk 8]
```diff
@@ -40,7 +40,15 @@ export default function Component({ service }) {
...
+            data={pool.data}
+            nasType={widget?.nasType}
```
Comment (shamoon) [inline src/widgets/truenas/component.jsx:50]:
```suggestion
            nasType={widget?.nasType ?? "scale"}
```

**Review by shamoon — CHANGES_REQUESTED** [review]
Great, yea agree thats much cleaner, thanks.

A couple more small things, which you might want to double-check syntax etc

**Comment by rgon10** [comment]
Pulled in the requested changes and validated the widget is still working as expected with Core. 
Thank you for the assistance. 
Let me know if you see anything else!

**Review by shamoon — APPROVED** [review]
Awesome, thanks.

Screenshot 2024-04-01 at 2 29 16 PM
Screenshot 2024-04-01 at 2 28 58 PM

**Comment by aroundmyroom** [comment]
How can I check why this feature does not work for me?
it only shows still the systemload, uptime and alerts

I have the commit active and available .. 

commit 4e69ea6088839c19a7242a4d05d97ab207927ced
Author: rgon10 <3789272+rgon10@users.noreply.github.com>
Date:   Mon Apr 1 17:32:39 2024 -0400

    Fix: TrueNAS Core support for pool stats (#3206)

my services.yaml shows 

- Shark NAS:
         href: http://nas.local.net
         description: NAS
         icon: truenas.png
         widget:
           type: truenas
           url: http://nas.local.net
           key: xxxxxxxdllwFUy6pgVAQW5rkg2xxxxx
           enablePools: true
           nasType: core
           # username: xxx # not required if using api key
           # password: xxx # not required if using api key

When saved no errors, but no pools available..

**Comment by shamoon** [comment]
Please dont post support requests on a closed PR, open a discussion. In general, the more information (logs, etc) you provide the better.

**Pull request merged**